app.js

```diff
@@ -0,0 +1,29 @@
+var app = require('express')();
+var http = require('http').Server(app);
+var io = require('socket.io')(http);
+var path = require('path');
+
+app.get('/', function(req, res) {
+	res.sendFile(path.join(__dirname, 'index.html'));
+});
+
+var users = [];
+
+io.on('connection', function(socket) {
+
+	socket.on('set username', function(name) {
+		//username not taken
+		if(users.indexOf(name) == -1) {
+			socket.emit('user set', name);
+		}	
+		//username taken
+		else {
+			socket.emit('user exists', name);
```

Comment:
Why do you want to remove it?
It would be someone else's username so we want it in users array!